Plutarch/Internal/PlutusType.hs

```diff
@@ -89,6 +90,12 @@ import Plutarch.Internal.Term (PType, S, Term, pdelay, pforce, plam', plet, puns
 import Plutarch.Internal.Witness (witness)
 import PlutusCore qualified as PLC
 
+type family PInnerMost' (a :: S -> Type) (b :: S -> Type) :: S -> Type where
+  PInnerMost' a a = a
+  PInnerMost' a _b = PInnerMost' (PInner a) a
+
+type PInnerMost a = PInnerMost' (PInner a) a
```

Comment:
I would spell it `PInnermost` - the original word is 'innermost', not 'inner most'.